pyomo/contrib/alternative_solutions/obbt.py

```diff
@@ -0,0 +1,370 @@
+            opt.options[parameter] = value
+        try:
+            results = opt.solve(
+                model, warmstart=warmstart, tee=tee, load_solutions=False
+            )
+        except:
+            # Assume that we failed b.c. of warm starts
+            results = None
+        if results is None:
+            results = opt.solve(model, tee=tee, load_solutions=False)
+        condition = results.solver.termination_condition
+        optimal_tc = pe.TerminationCondition.optimal
+        infeas_or_unbdd_tc = pe.TerminationCondition.infeasibleOrUnbounded
+        unbdd_tc = pe.TerminationCondition.unbounded
+    if not quiet:  # pragma: no cover
+        print("Performing initial solve of model.")
+
+    if condition != optimal_tc:
+        raise RuntimeError(
+            ("OBBT cannot be applied, " "TerminationCondition = {}").format(
+                condition.value
+            )
+        )
+    if use_appsi:
+        results.solution_loader.load_vars(solution_number=0)
+    else:
+        model.solutions.load_from(results)
+    if warmstart:
+        _add_solution(solutions)
+    orig_objective_value = pe.value(orig_objective)
+    if not quiet:  # pragma: no cover
+        print("Found optimal solution, value = {}.".format(orig_objective_value))
+    aos_block = aos_utils._add_aos_block(model, name="_obbt")
+    if not quiet:  # pragma: no cover
+        print("Added block {} to the model.".format(aos_block))
+    obj_constraints = aos_utils._add_objective_constraint(
+        aos_block, orig_objective, orig_objective_value, rel_opt_gap, abs_opt_gap
+    )
+    new_constraint = False
+    if len(obj_constraints) > 0:
+        new_constraint = True
+    orig_objective.deactivate()
+
+    if use_appsi:
+        opt.update_config.check_for_new_or_removed_constraints = new_constraint
+        opt.update_config.check_for_new_or_removed_vars = False
+        opt.update_config.check_for_new_or_removed_params = False
+        opt.update_config.check_for_new_objective = True
+        opt.update_config.update_constraints = False
+        opt.update_config.update_vars = False
+        opt.update_config.update_params = False
+        opt.update_config.update_named_expressions = False
+        opt.update_config.update_objective = False
+        opt.update_config.treat_fixed_vars_as_params = False
+
+    variable_bounds = pe.ComponentMap()
+    solns = [Solution(model, all_variables, objective=orig_objective)]
+
+    senses = [(pe.minimize, "LB"), (pe.maximize, "UB")]
+
+    iteration = 1
+    total_iterations = len(senses) * num_vars
+    for idx in range(len(senses)):
+        sense = senses[idx][0]
+        bound_dir = senses[idx][1]
+
+        for var in variable_list:
+            if idx == 0:
+                variable_bounds[var] = [None, None]
+
+            if hasattr(aos_block, "var_objective"):
+                aos_block.del_component("var_objective")
+
+            aos_block.var_objective = pe.Objective(expr=var, sense=sense)
```

Comment:
Pyomo will let you assign to `var_objective.expr` and `var_objective.sense`, so you don't have to add and remove the component every iteration if you don't want.

Reply:
Really?  I thought Pyomo complained when you re-added a component to a model.  This logic avoids those warnings.

Reply:
It will complain if you replace a component with another component.  What @emma58 was suggesting was to just assign a new expression on the existing Objective.  That is, 
```python
aos_block.var_objective.expr = var
aos_block.var_objective.sense = sense
```

Reply:
Done, but really it's not more parsimonious b.c. you need an if/else statement to cover first time you add the objective.

Reply:
Since it's on the `aos_block`, you could just make the objective when you originally make the Block and then not need the if/else since you'll know it's already there.